### DISCUSSION TITLE
mpp-solar on windows 7

### DISCUSSION THREAD

**Discussion by excel4** [jblance/mpp-solar#466] [Q&A]
I would like to use mpp-solar to send commands and receive data from an mpp-solar inverter. I do not need the server elements - just the basic command-line interface. I am running on Windows 7. The readme says that version v0.16 requires python 3.10..0, but python 10.0 does not run on windows 7.

Will the mpp-solar command-line run on Windows 7 with python 3.8, the last version that supports Windows 7? 

If not, what process do I need to follow to install an older version of mpp-solar? I tried downloading the v.0.7.49 release but I could not get it to run. There is no mpp-solar.exe file in the zip file and I do not see instructions for building it on windows.

Thanks!

**Comment by Saentist** [comment]
Virtual PC is what you can install.
Then use Ubuntu server 22.04

There is no WSL for windows 7 to use it native 
Cygwin is some alternative

There is also MultiSIBControl but no so flexible as MPP-Solar

**Comment by PaulEPop** [comment]
Hi excel4, I would like to know, how you are using mpp-solar under windows cmd.
How do you specify the port for it to work?
mpp-solar -p **/dev/hidraw1** -P PI17 -c PS --filter "input_power|total_active" only works under linux
Thank you in advance

**Reply by excel4** [reply]
I have not been using it, since I could not get it to work properly. I am using watchpower, which I am have only been able to install in Windows 7. I was hoping to use both on the same hardware, but that does not appear to be possible.

**Reply by PaulEPop** [reply]
Hi,

at the moment I'm using a VM in VirtualBox with Raspian.
Maybe I can make an OVA file and share it for you, if you want.
But you have to decide, mpp-solar or watchpower, both is not possible at the same time.

**Reply by excel4** [reply]
Thanks! I'd like to test out mpp-solar, so a VM would help. So, if it is easy to post an OVA, that would be great. I am running watchpower in a VM under virtualbox, so adding another VM would be easy. I understand that both cannot be used at the same time.

**Reply by jblance** [reply]
there is also a docker image on dockerhub - https://hub.docker.com/r/jblance/mppsolar
mppsolar is not really suited to windows, it would need (at least) a new IO module to talk to Windows USB devices

**Reply by excel4** [reply]
OK - sounds like that won't work well for me since my Virtualbox host is Windows. What about using mpp-solar with a serial interface to the inverter instead of USB? I am currently using watchpower with a serial line through a prolific USB-serial adapter.

**Reply by jblance** [reply]
do you know which com port the connection is on?
is it direct or via a usb to rs232 adapter?

**Reply by jblance** [reply]
also have you managed to install python and pip on windows?

**Reply by excel4** [reply]
1) I am currently running watchpower on a serial port provided by a usb-serial adapter (prolific). Virtualbox provides the com port to the Windows 7 guest VM.
2) yes, I know the com port
3) I have installed python and pip on a windows 10 machine (not a VM).
4) However, I would prefer to run mpp-solar on a Raspian or Alpine linux VM guest vs. Windows guest VM.
5) I would be happy to run mpp-solar on the same Windows 7 VM that watchpower uses, but current versions of Python do not support Windows 7.

If you think that mpp-solar will run in an Alpine VM under Virtualbox Windows host using a mapped serial port, I could try that. Or if you think the Rapsian OVA will work, I could try that.

**Comment by jblance** [comment]
It will run on a Linux VM and should definitely run Alpine
Try it and let me know if you get issues
Win7 is pretty dated, maybe time to install a Linux distro 😅

**Comment by excel4** [comment]
OK. FYI - my VM host is Windows 10. I was not able to get watchpower to run properly on Windows 10 or Linux, hence the Windows 7 VM.

**Reply by PaulEPop** [reply]
Okay? On my system (Win10 Host), watchpower is running well (watchPower is a piece of creepy software).
I'm using Raspbian on VirtualBox and the python tool from jblance, it works perfect!
I will try to add the php script from riogrande into this VM. If I'm finnished, I will provide my VM via Onedrive.
In this case you have to change the config files only and start the mpp deamon.

**Reply by excel4** [reply]
Thanks. The issue I have with Win10 is related to strange fonts during setup. I tried a number of things based on web searches, but I could not get it to work properly either in a VM or directly. Win7 works fine so I stayed with that.

I am confused on one point: You are running mpp-solar in a raspian VM under a Win10 host. That sounds like it would work well for me.  Bu jblance said: "mppsolar is not really suited to windows, it would need (at least) a new IO module to talk to Windows USB devices".

How did you resolve this?

**Reply by PaulEPop** [reply]
![20240604_224233](https://github.com/jblance/mpp-solar/assets/153681180/bd52ac7d-4e73-499a-b55c-cf3f10bdce00)

I have two inverters MPP-Solar 12k

**Reply by Saentist** [reply]
@PaulEPop try http://www.multisibcontrol.net/ better then watch power on windows

**Reply by PaulEPop** [reply]
> @PaulEPop try http://www.multisibcontrol.net/ better then watch power on windows

I'm using the python tool from jblance and send all information I want to have to mqtt / Homeassistant.
For configurationchanges I'm using the App on the phone. I was trying the multisib, but it isn't a solution for me.

**Reply by excel4** [reply]
Thanks for the pointer. I had not run across that tool. I'm hesitant since I can't find any reviews of it, information on the company or person developing and supporting it, etc.

**Comment by jblance** [comment]
It won't matter of you're running it in a VM. It's the OS interface that
matters (ie the one in the VM)

TBH it would probably work natively in Win10 but I've never tried (I
actually wrote it as my inverter is in a shed and I didn't want to put a PC
in there)

**Reply by PaulEPop** [reply]
I'm using a small NUC with win10 + RealVNC for remote control
On this host, I have running Homeassistant as VM + your great tool in a 2nd VM.
